packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1311,4 +1319,304 @@ export class KubernetesClient {
       this.telemetry.track('kubernetesExecIntoContainer', telemetryOptions);
     }
   }
+
+  async restartPod(name: string): Promise<void> {
+    let telemetryOptions = {};
+    try {
+      const ns = this.currentNamespace;
+      const connected = await this.checkConnection();
+      if (!ns) {
+        throw new Error('no active namespace');
+      }
+      if (!connected) {
+        throw new Error('not active connection');
+      }
+
+      const pod = await this.readNamespacedPod(name, ns);
+      if (!pod?.metadata) {
+        throw new Error('no metadata found');
+      }
+
+      const creationSource = this.checkPodCreationSource(pod.metadata);
+      if (creationSource.isManuallyCreated) {
+        await this.restartManuallyCreatedPod(name, ns, pod);
+      } else {
+        if (!creationSource.controllerType) {
+          throw new Error('unable to restart controlled pod');
+        }
+
+        const controller = this.getPodController(pod.metadata);
+        const controllerName = controller!.name;
+
+        if (['Deployment', 'ReplicaSet', 'StatefulSet'].includes(creationSource.controllerType)) {
+          const controllerType: 'Deployment' | 'ReplicaSet' | 'StatefulSet' = creationSource.controllerType as
+            | 'Deployment'
+            | 'ReplicaSet'
+            | 'StatefulSet';
+          await this.scaleControllerToRestartPods(ns, controllerName, controllerType);
```

Comment:
```suggestion
       ```suggestion
        if (isScalableControllerType(creationSource.controllerType)) {
          await this.scaleControllerToRestartPods(ns, controllerName, creationSource.controllerType)
```

Reply:
updated